src/cbapi/psc/livequery/models.py

```diff
@@ -45,12 +47,70 @@ def _query_implementation(cls, cb):
         return RunQuery(cls, cb)
 
     def _refresh(self):
-        url = self.urlobject.format(self._cb.credentials.org_key, self.id)
+        if self._is_deleted:
+            raise ApiError("cannot refresh a deleted query")
+        url = self.urlobject_single.format(self._cb.credentials.org_key, self.id)
         resp = self._cb.get_object(url)
         self._info = resp
         self._last_refresh_time = time.time()
         return True
+    
+    def stop(self):
+        if self._is_deleted:
+            raise ApiError("cannot stop a deleted query")
+        url = self.urlobject_single.format(self._cb.credentials.org_key, self.id) + "/status"
+        result = self._cb.put_object(url, {'status': 'CANCELLED'})
+        if (result.status_code == 200):
+            try:
+                self._info = result.json()
+                self._last_refresh_time = time.time()
+                return True
+            except:
+                raise ServerError(result.status_code, "Cannot parse response as JSON: {0:s}".format(result.content))            
+        return False
+    
+    def delete(self):
+        if self._is_deleted:
+            return True # already deleted
+        url = self.urlobject_single.format(self._cb.credentials.org_key, self.id)
+        result = self._cb.delete_object(url)
+        if result.status_code == 200:
+            self._is_deleted = True
+            return True
+        return False
+    
+    def query_device_summaries(self):
```

Comment:
Why are we adding `query_device_summaries`, `query_result_facets`, and `query_device_summary_facets` here since these don't query the run but instead query the result object. They are helpful functions but maybe move them to the result model?

Reply:
They work on the run ID, and you don't necessarily need to have queried results to use them. Maybe they should be in _both_ places?

Reply:
The path doesn't utilize `/run` so you are not querying the run object but instead `/result`. The result object is tied to the run object by the id as they share them same id. This will also flow better with the examples Will has created for CBAPI since you need to query the result `results = cb.select(Result).run_id(args.id)` after a run as the run may have not finished and you will want to be able to refresh the results

Reply:
You do make a compelling point.  There may be multiple Result objects for one Run, but you'd then be able to run these three queries on any of them, as they all have the same run_id.  I can make that change.